### ISSUE TITLE
Experiments should leave behind json logs

### ISSUE THREAD

**Issue by sohaibimran7** [AISC10-team09/bayesian_reasoning#2]
These logs should have at least, 

an id, 
the class pairs c1 and c2,
the data X, 
the conversation history h, 
the prompt template, 
the actual prompt used (unless exactly reconstructable via the above),
the model, 
model parameters,
selected logit at each autoregressive step, and it's logprob
Toplogprobs (both logits and logprobs) at each autoregressive step, (100 or vocab_size based on a flag/parameter) 
Cumulative logprob of the selected logits (or the probability that translates to).
